wee_slack.py

```diff
@@ -2938,20 +2938,51 @@ def msg_command_cb(data, current_buffer, args):
 @slack_buffer_or_ignore
 def command_talk(data, current_buffer, args):
     """
-    Open a chat with the specified user
-    /slack talk [user]
+    Open a chat with the specified user(s)
+    /slack talk [user] ([user2] [user3]...)
     """
 
     data = decode_from_utf8(data)
     args = decode_from_utf8(args)
     e = EVENTROUTER
     team = e.weechat_controller.buffers[current_buffer].team
-    channel_name = args.split(' ')[1]
+    users = args.split(' ')[1:]
     c = team.get_channel_map()
+
+    if len(users) > 1:
+        # Add the current user if not specified since all MPDM names have it
+        if team.nick not in users:
+            users.append(team.nick)
+
+        # Use the user order given if no channels are found below
+        channel_name = 'mpdm-{}-1'.format('--'.join(users))
+
+        # Use the name of any existing channel with the same set of users
+        user_set = set(users)
+        mpdm_regex = re.compile(r'-+')
+        for channel in c:
+            if channel.startswith('mpdm-') and user_set == set(
+                    mpdm_regex.split(channel)[1:-1]):
+                channel_name = channel
+                break
+    else:
+        channel_name = users[0]
+
     if channel_name not in c:
+        # Get the IDs of the users
         u = team.get_username_map()
-        if channel_name in u:
-            s = SlackRequest(team.token, "im.open", {"user": u[channel_name]}, team_hash=team.team_hash)
+        user_ids = tuple(u[user] for user in users if user in u)
+
+        # Open the DM or MPDM depending on the number of users
+        if user_ids:
+            if len(user_ids) > 1:
+                method = 'mpim.open'
```

Comment:
We should probably use the `SLACK_API_TRANSLATOR` mapping here (though I see it wasn't used before either, not sure why).

Reply:
Good call, especially since that PR is switching to use the Conversations API for open methods.

Reply:
So it turns out only mpim.open and conversations.open work for MPDMs, and those aren't in `SLACK_API_TRANSLATOR` currently. Might be better to hold off updating this until #451 is merged in that case, it'd also simplify `params` since conversations.open accepts `"users"` for both DMs and MPDMs.